spikeextractors/extractors/nwbextractors/nwbextractors.py

```diff
@@ -455,141 +456,148 @@ def add_electrodes(recording: se.RecordingExtractor, nwbfile=None,
-                                  " with 'data' being a list of items!" +
-                                  " The custom column will not be added.")
-                else:
-                    metadata_columns = []
-                    print("Warning: metadata must be a list of dictionaries of the form"
-                          " metadata['Ecephys']['Electrodes'] = [{'name': my_name,"
-                          " 'description': my_description, 'data': [my_electrode_data]}, ...]"
-                          " where [my_electrode_data] is a list in one-to-one correspondence with"
-                          " the nwbfile electrode ids and RecordingExtractor channel ids!")
+        if nwbfile.electrodes is None:
+            nwb_elec_ids = []
+        else:
+            nwb_elec_ids = nwbfile.electrodes.id.data[:]
+        for metadata_column in metadata['Ecephys']['Electrodes']:
+            if metadata_column['name'] not in nwbfile.electrodes.colnames:
+                nwbfile.add_electrode_column(str(metadata_column['name']),
+                                             str(metadata_column['description']))
 
-        for j, channel_id in enumerate(channel_ids):
-            # Will not overwrite the electrode id if already in nwbfile
+        for j, channel_id in enumerate(recording.get_channel_ids()):
             if channel_id not in nwb_elec_ids:
-                electrode_kwargs = fill_kwargs_from_defaults(defaults)
-                electrode_kwargs.update({'id': channel_id})
+                electrode_kwargs = dict(defaults)
+                electrode_kwargs.update(id=channel_id)
 
-                # recording.get_channel_locations defaults to np.nan is there are none
+                # recording.get_channel_locations defaults to np.nan if there are none
                 location = recording.get_channel_locations(channel_ids=channel_id)[0]
-                if location[0] is not np.nan or location[1] is not np.nan:
-                    electrode_kwargs.update({'rel_x': float(location[0]),
-                                             'rel_y': float(location[1])})
-
-                for metadata_column in metadata_columns:
-                    if type(metadata_column) is dict \
-                            and set(metadata_column.keys()) == set(['name', 'description', 'data']) \
-                            and type(metadata_column['data']) is list:
-                        if metadata_column['name'] == 'group':
-                            if list_get(metadata_column['data'], j, np.nan) not in nwbfile.electrode_groups:
-                                print(
-                                    f"Warning: Electrode group for electrode {channel_id} was not found in the nwbfile."
-                                    " Automatically adding!")
-                                se.NwbRecordingExtractor.add_electrode_groups(recording, nwbfile, metadata)
+                if any([loc is np.nan for loc in location]):
```

Comment:
@CodyCBakerPhD Here we want both elements of `location` not to be `nan` to update the dictionary.

Also `loc is np.nan` is not the correct way to test for nans. Try this snippet:
```
location = np.array([np.nan, np.nan])
print(location[0] is np.nan)  # --> prints False...
print(np.isnan(location[0])  # --> correctly prints True
```

```suggestion
                if all([not np.isnan(loc) for loc in location]):
```

Reply:
Interestingly, if you remove the `np.array` converion on the list of nans, it works as expected; but I did not realize that `get_channel_locations()` returns things as `np.array`'s, so definitely implementation that then.